dotnet/src/webdriver/WebDriver.cs

```diff
@@ -556,7 +557,25 @@ internal ReadOnlyCollection<IWebElement> GetElementsFromResponse(Response respon
         /// <returns>WebDriver Response</returns>
         internal Response InternalExecute(string driverCommandToExecute, Dictionary<string, object> parameters)
         {
-            return this.Execute(driverCommandToExecute, parameters);
+            return AsyncHelper.RunSync(() => this.InternalExecuteAsync(driverCommandToExecute, parameters));
+        }
+
+        /// <summary>
+        /// Executes commands with the driver asynchronously
+        /// </summary>
+        /// <param name="driverCommandToExecute">Command that needs executing</param>
+        /// <param name="parameters">Parameters needed for the command</param>
+        /// <returns>A task object representing the asynchronous operation</returns>
+        internal Task<Response> InternalExecuteAsync(string driverCommandToExecute,
```

Comment:
If I got the idea of `InternalExecuteAsync` right, it is just a 
copy of `ExecuteAsync` but with `internal` access modifier. If so, let's maybe change the access modifier of `ExecuteAsync` from `protected` to `protected internal`, and then `InternalExecuteAsync` can be removed.

> A `protected internal` member is accessible from the current assembly or from types that are derived from the containing class.